src/TodoApp.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useState, useEffect, useMemo } from 'react';
+import { TodoList } from './components/TodoList';
+import { TodosFilter } from './components/TodosFilter';
+import { FILTERS } from './constants';
+
+const getfilteredTodos = (items, status) => {
+  switch (status) {
+    case FILTERS.all:
+      return items;
+    case FILTERS.active:
+      return items.filter(item => !item.completed);
+    case FILTERS.completed:
+      return items.filter(item => item.completed);
+    default:
+      return 'Error!';
+  }
+};
+
+const TodoApp = () => {
+  const [todos, setTodos] = useState([]);
+  const [newTodo, setNewTodo] = useState('');
+  const [filter, setFilter] = useState('all');
+  const completedTodos = todos.filter(todo => !todo.completed).length;
+
+  useEffect(() => {
+    if (!localStorage.todos) {
+      localStorage.setItem('todos', JSON.stringify([]));
+    } else {
+      setTodos(JSON.parse(localStorage.getItem('todos')));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  }, [todos]);
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!newTodo) {
+      return;
+    }
+
+    setTodos([
+      ...todos,
+      {
+        id: +new Date(),
+        title: newTodo,
+        completed: false,
+      },
+    ]);
+
+    setNewTodo('');
+  };
+
+  const updateTodoStatus = (todoId) => {
+    setTodos(todos.map((todo) => {
+      if (todo.id !== todoId) {
+        return { ...todo };
+      }
+
+      return {
+        ...todo,
+        completed: !todo.completed,
+      };
+    }));
+  };
+
+  const updateTodoTitle = (todoId, newTitle) => {
+    setTodos(todos.map((todo) => {
+      if (todo.id !== todoId) {
+        return { ...todo };
+      }
+
+      return {
+        ...todo,
+        title: newTitle,
+      };
+    }));
+  };
+
+  const toogleAll = (event) => {
+    setTodos(todos.map(todo => (
+      {
+        ...todo,
+        completed: event.target.checked,
+      }
+    )));
+  };
+
+  const filteredTodos = useMemo(
+    () => getfilteredTodos(todos, filter),
+    [todos, filter],
+  );
+
+  const deleteTodo = (todoId) => {
+    setTodos(todos.filter(todo => todo.id !== todoId));
+  };
+
+  const clearCompleted = () => {
+    setTodos(todos.filter(todo => !todo.completed));
+  };
```

Comment:
use useCallback for functions

Reply:
what for? I don't use pass these functions as deps in other hooks